patroni/postgresql.py

```diff
@@ -521,6 +525,26 @@ def _initialize(self, config):
             self.set_state('initdb failed')
         return ret
 
+    def _custom_bootstrap(self, config):
+        params = ['--scope=' + self.scope, '--datadir=' + self._data_dir]
+        try:
+            logger.info('Running custom bootstrap script: %s', config['command'])
+            if subprocess.call(shlex.split(config['command']) + params) != 0:
+                self.set_state('custom bootstrap failed')
+                return False
+        except Exception:
+            logger.exception('Exception during custom bootstrap')
+            return False
+
+        self._post_restore()
+        self.save_configuration_files()
```

Comment:
What if we don't have a `.backup` files? We really need to have them, because we are going to change `pg_hba.conf` and should be able to restore the old version.